### PR TITLE
NPM-only components

### PR DESCRIPTION
**Pull request by notlee**
This is the beta branch. Changes were reviewed here: https://github.com/Financial-Times/create-origami-component/pull/235 (and elsewhere)

### CODE REVIEW

**Comment by notlee** [comment]
Percy shows the icons are missing because the image service hasn't released yet

File: templates/npmignore.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+/node_modules/
+/build/
+.idea/
+/demos/local
+/coverage
+.eslintrc.js
+.github
+.gitignore
+.remarkrc.js
+.stylelintrc.js
```
Comment (JakeChampion) [inline templates/npmignore.js:15]:
```suggestion
.eslintrc.cjs
.github/
.gitignore
.remarkrc.cjs
.stylelintrc.cjs
```

File: templates/origami-manifest.js [hunk 2]
```diff
@@ -1,10 +1,8 @@
 const details = answers => {
 	return `{
-	"description": "${answers.description}",
-	"keywords": ${JSON.stringify(answers.keywords)},
 	"origamiType": "module",
 	"origamiCategory": "${answers.category}",
-	"origamiVersion": 1,
+	"origamiVersion": 2,
```
Comment (JakeChampion) [inline templates/origami-manifest.js:5]:
```suggestion
	"origamiVersion": "2.0",
```

File: templates/package-json.js [hunk 3]
```diff
@@ -1,16 +1,69 @@
...
+	if (answers.javascript) {
+		Object.assign(packageJson, {
+			"exports": {
+				".": {
+					"browser": "./main.js"
+				}
+			},
+			"browser": "./main.js",
+			"types": "./types/main.d.ts"
+		});
```
Comment (JakeChampion) [inline templates/package-json.js:41]: Let's remove the typescript parts for now, we haven't added it to any of the spec v2 components we are currently working on
```suggestion
		Object.assign(packageJson, {
			"browser": "./main.js",
		});
```

File: templates/package-json.js [hunk 4]
```diff
@@ -1,16 +1,69 @@
...
+
+		Object.assign(packageJson.scripts, {
+			"build": "tsc src/**/*.js main.js --declaration --allowJs --emitDeclarationOnly --outDir types",
+			"version": "npm run build"
+		});
```
Comment (JakeChampion) [inline templates/package-json.js:46]:
```suggestion
```

File: templates/package-json.js [hunk 5]
```diff
@@ -1,16 +1,69 @@
...
+
+		Object.assign(packageJson.devDependencies, {
+			"eslint": "^7.19.0",
+			"eslint-config-origami-component": "^2.1.0",
+			"typescript": "^4.0.5"
```
Comment (JakeChampion) [inline templates/package-json.js:51]:
```suggestion
			"eslint-config-origami-component": "^2.1.0"
```

**Comment by notlee** [comment]
This released as `v0.3.2` since I used the patch release label but the existing beta tag does not reference this branch.

File: package.json [hunk 6]
```diff
@@ -19,7 +19,7 @@
     "jest": "^26.0.0",
     "mocha": "^8.0.1",
     "nixt": "github:vesln/nixt#d2075099ecf935c870705fa5d5de437113d4d272",
-    "origami-build-tools": "^10.5.9",
+    "origami-build-tools": "prerelease",
```
Comment (JakeChampion) [inline package.json:22]:
```suggestion
    "origami-build-tools": "^11",
```

File: templates/package-json.js [hunk 7]
```diff
@@ -1,16 +1,67 @@
...
+			"type": "git",
+			"url": `git+https://github.com/Financial-Times/${answers.name}.git`
+		},
+		"license": "MIT",
+		"author": "",
+		"type": "module",
+		"devDependencies": {
+			"remark": "^12.0.1",
+			"remark-lint": "^7.0.1",
+			"remark-preset-lint-origami-component": "^2.0.0-beta.1"
```
Comment (JakeChampion) [inline templates/package-json.js:22]:
```suggestion
			"remark-preset-lint-origami-component": "^2.0.0"
```

File: templates/package-json.js [hunk 8]
```diff
@@ -1,16 +1,67 @@
...
+	if (answers.scss) {
+		Object.assign(packageJson.devDependencies, {
+			"stylelint": "^13.9.0",
+			"stylelint-config-origami-component": "^1.0.4"
+		});
+	}
+
+	if (answers.brands) {
+		packageJson.peerDependencies = {
+			"@financial-times/o-brand": "^4.0.0-beta.0"
```
Comment (JakeChampion) [inline templates/package-json.js:50]:
```suggestion
			"@financial-times/o-brand": "^4.0.0"
```

File: templates/workflows/test-origami-component.yml [hunk 9]
```diff
@@ -5,18 +5,11 @@ jobs:
...
-      id: repo_name
-      run: echo ::set-output name=REPO_NAME::${GITHUB_REPOSITORY/*\//}
-    - run: npx origami-ci branch
-      env:
-        ORIGAMI_CI_NAME: ${{ steps.repo_name.outputs.REPO_NAME }}
-        ORIGAMI_CI_VERSION: 0.0.0
-        GITHUB_TOKEN: ${{ secrets.ORIGAMI_FOX_ACCESS_TOKEN }}
-        ORIGAMI_PULL_REQUEST: ${{ github.event.pull_request.html_url }}
+    - run: npm install -g npm@^7
+    - run: npm install -g origami-build-tools@prerelease
```
Comment (JakeChampion) [inline templates/workflows/test-origami-component.yml:12]:
```suggestion
    - run: npm install -g origami-build-tools@^11
```

**Review by JakeChampion — DISMISSED** [review]

File: templates/workflows/test-origami-component.yml (hunk 9 above)
Comment (JakeChampion) [reply templates/workflows/test-origami-component.yml]: fixed in [`7733640` (#238)](https://github.com/Financial-Times/create-origami-component/pull/238/commits/7733640bb2ff9c761fd2babfa3fe5e8260d1db84)

File: package.json (hunk 6 above)
Comment (JakeChampion) [reply package.json]: fixed in [`7733640` (#238)](https://github.com/Financial-Times/create-origami-component/pull/238/commits/7733640bb2ff9c761fd2babfa3fe5e8260d1db84)

File: templates/package-json.js (hunk 8 above)
Comment (JakeChampion) [reply templates/package-json.js]: fixed in [`1b0e344` (#238)](https://github.com/Financial-Times/create-origami-component/pull/238/commits/1b0e3449b950710fc2defeaab53eac4faeb36ea9)

File: templates/package-json.js (hunk 7 above)
Comment (JakeChampion) [reply templates/package-json.js]: fixed in [`1b0e344` (#238)](https://github.com/Financial-Times/create-origami-component/pull/238/commits/1b0e3449b950710fc2defeaab53eac4faeb36ea9)

**Review by chee — APPROVED** [review]

**Comment by origamiserviceuser** [comment]
:tada: This PR is included in version v1.0.0 :tada:

**Pull request merged**